projects/plugins/jetpack/modules/comments/comments.php

```diff
@@ -479,104 +479,53 @@ public function watch_comment_parent() {
 			// @todo Implement AMP support.
 			return;
 		}
-
-		$url_origin = 'https://jetpack.wordpress.com';
 		?>
-
-		<!--[if IE]>
-		<script type="text/javascript">
-			if ( 0 === window.location.hash.indexOf( '#comment-' ) ) {
-				// window.location.reload() doesn't respect the Hash in IE
-				window.location.hash = window.location.hash;
-			}
-		</script>
-		<![endif]-->
 		<script type="text/javascript">
-			(function () {
-				var comm_par_el = document.getElementById( 'comment_parent' ),
-					comm_par = ( comm_par_el && comm_par_el.value ) ? comm_par_el.value : '',
-					frame = document.getElementById( 'jetpack_remote_comment' ),
-					tellFrameNewParent;
-
-				tellFrameNewParent = function () {
-					if ( comm_par ) {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>" + '&replytocom=' + parseInt( comm_par, 10 ).toString();
-					} else {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>";
-					}
-				};
-
-				<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
-
-				if ( 'undefined' !== typeof addComment ) {
+			<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
+			const iframe = document.querySelector( '#jetpack_remote_comment' );
```

Comment:
As written `iframe` will not be defined down on line 526 if "thread_comments" is off. We need to move that line outside of the `<?php if`.
```suggestion
			const iframe = document.querySelector( '#jetpack_remote_comment' );
			<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
```

Reply:
Addressed!